core/Date.php

```diff
@@ -611,6 +612,38 @@ public function subYear($n)
         return new Date($result, $this->timezone);
     }
 
+    protected static $use12HourClock;
+
+    protected function getTimeFormat()
+    {
+        if (is_null(self::$use12HourClock)) {
+            self::$use12HourClock = LanguagesManager::uses12HourClockForCurrentUser();
```

Comment:
Ok, I don't think it would be a problem here if it changes during the request but for some things it can be a problem for sure.